java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -394,6 +395,24 @@ public static ColumnVector makeStruct(long rows, ColumnView... columns) {
     }
   }
 
+  /**
+   * This is a very specialized method that has only one job. It takes in a list and returns a
```

Comment:
It feels almost too specialized. Do you have performance numbers comparing doing this vs using something like `replaceColumnsInStruct`, along with something similar for lists?  Perhaps we could have versions of them that return views instead of a ColumnVector so users can build up the nested structure that they want, and then at the end copy it out?

What if I have a list of structs with lists of structs in it and one of the nodes is a decimal32?

Reply:
I don't have performance numbers but I started off with the approach you are suggesting namely a method taking in a d64 column and returning a d32 column_view but I faced a number of problems. First, the column_view doesn't own the underlying buffers so we have to copy to a column_vector before getting the `DeviceMemoryBuffer` required for constructing a List column_vector. Secondly, if we have multiple layers of nesting in a List, e.g. 

List<List<D32>> 
the method will first cast D64 to a D32  column_view, then copy the offsets from the original vector to a completely new column_vector and shove this D32 vector in it. Then we need to do the whole thing again for the top layer List. I just thought this would cause unnecessary copying of items when the offsets don't change at all as long as the row count is the same, which it is in this case. Even if we opt to return a column_vector from the method casting D64 to D32 we still have to copy the offsets unnecessarily. 

I agree this is very specialized but I couldn't think of a better way short of making a cudf change to allow me to swap a child column which I can only imagine will be a much bigger and tougher battle to fight. May be I am missing something that is obvious to you, please share

Reply:
> First, the column_view doesn't own the underlying buffers so we have to copy to a column_vector before getting the DeviceMemoryBuffer required for constructing a List column_vector

What I am suggesting is that you build up a column view so it looks exactly how you want it to (with no data being copied at all), and then you copy the data out when it is all done. As you have it right now, the C++ code is implicitly copying the data at each level of the recursion. `cudf::column* replace_column(cudf::column list_column) {` takes a `cudf::column` as an argument.  This means it will make a copy of that data implicitly as a part of the call. This is very inefficient and not flexible enough to do what you really want it to.

So as an alternative to this code I would like to see an API in ColumnView for list like.

```
ColumnView replaceChildWithView(int index, ColumnView view)
```

Then you could have code like the following.  It would allow you to do all of the view manipulations you want, lists , structs, lists of structs or lists of structs, with only copying the data needed until you are done and can copy everything for the final result.

```
try (ColumnVector original = ...;
    ColumnView d64Column = original.getChildColumnView(1);
    ColumnVector d32Column = d64Column.cast...;
    ColumnView newView = original.replaceChildWithView(1, d32Column)) {
    return newView.copyToColumnVector();
}
```

Reply:
Thank you for this Bobby. I was over looking the fact that I don't need to replace a child in a view, I can just create a new one. I have made the changes, the code looks much better. Thanks